File: services/frontend/src/lib/schemas/team-llm-api-keys.ts
import z from 'zod';

export const createTeamLLMAPIKeyInput = z.object({
  provider: z.string().min(1, 'Provider is required').max(50),
  api_key: z.string().min(1, 'API key is required'),
});

export type CreateTeamLLMAPIKeyInput = z.infer<
  typeof createTeamLLMAPIKeyInput
>;

export const saveAPIKeyFormSchema = createTeamLLMAPIKeyInput.extend({
  team_id: z.string().transform((val) => parseInt(val, 10)),
});

export type SaveAPIKeyFormData = z.infer<typeof saveAPIKeyFormSchema>;

export const teamLLMAPIKeyStatusResponse = z.object({
  id: z.number(),
  provider: z.string(),
  is_active: z.boolean(),
  created_at: z.string(),
  updated_at: z.string(),
  last_used_at: z.string().optional().nullable(),
});

export type TeamLLMAPIKeyStatusResponse = z.infer<
  typeof teamLLMAPIKeyStatusResponse
>;

export const teamLLMAPIKeysListResponse = z.array(teamLLMAPIKeyStatusResponse);

export type TeamLLMAPIKeysListResponse = z.infer<
  typeof teamLLMAPIKeysListResponse
>;

File: services/acacia/packages/schemas/team_llm_api_keys.go
package schemas

import (
	"errors"
	"time"

	"github.com/go-playground/validator/v10"
)

type CreateTeamLLMAPIKeyInput struct {
	Provider string `json:"provider" validate:"required,min=1,max=50"`
	APIKey   string `json:"api_key" validate:"required,min=1"`
}

type UpdateTeamLLMAPIKeyInput struct {
	APIKey string `json:"api_key" validate:"required,min=1"`
}

type TeamLLMAPIKeyStatusResponse struct {
	ID         int64      `json:"id"`
	Provider   string     `json:"provider"`
	IsActive   bool       `json:"is_active"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`
	LastUsedAt *time.Time `json:"last_used_at,omitempty"`
}

type TeamLLMAPIKeysListResponse []TeamLLMAPIKeyStatusResponse

// HandleTeamLLMAPIKeyValidationErrors converts validator errors to user-friendly messages
func HandleTeamLLMAPIKeyValidationErrors(err error) error {
	validationErrors, ok := err.(validator.ValidationErrors)
	if !ok {
		return errors.New("Validation failed")
	}

	for _, e := range validationErrors {
		switch e.Field() {
		case "Provider":
			if e.Tag() == "required" {
				return errors.New("Provider is required")
			}
			return errors.New("Provider must be between 1 and 50 characters")
		case "APIKey":
			if e.Tag() == "required" {
				return errors.New("API key is required")
			}
			return errors.New("API key cannot be empty")
		default:
			return errors.New("Validation failed")
		}
	}

	return errors.New("Validation failed")
}
